### ISSUE TITLE
Add task creation

### ISSUE THREAD

**Issue by joshuafielding** [joshuafielding/habitHeatMap#7]
Task creation needs to have:
  1. Title
  2. Description
  3. Date + Repeat options (ex: repeat every monday)

Task creation prompt:
[ title ]
[ desc. ]
[ 10/3/2024 ]
[ repeat settings ]

Example tasks when interacting with a day:
10/3/2024
  [✔️] Finish homework - cryptography
  [     ] Work on habitHeatMap - finish color customization

Could possibly add times for timeblocking, not sure if I'm going to. I wanted to have this primarily be a way to visualize progress towards certain goals instead of being a calendar.

**Comment by joshuafielding** [comment]
added html for task creation prompt, now need to append tasks to days.

**Comment by joshuafielding** [comment]
can append tasks to days, just need to be able to save tasks. probably going to do local storage for simplicity.

**Comment by joshuafielding** [comment]
local storage done.
current solution to adding tasks to days:
store task in task array, when day is clicked loop through array to find if task date = day.id.

need to find a way to have tasks that repeat on certain days.

**Comment by joshuafielding** [comment]
repeated tasks are working (kind of...)
the current solution is to append an attribute to each day div. data-dayname is a value of either Monday, Tuesday... Sunday. If one of those day names matches the day name of a task we append it to day pop up.

current problems:
* make it so when a user completes the task it doesn't complete all future ones.
* when a user wants something to repeat every 2 or 3 weeks/months/years it doesn't work.

**Comment by joshuafielding** [comment]
Weekly frequencies are working as expected!
The solution is quite complicated though.
In order to calculate what days are "valid" for our frequency we do the following:
1. We first create a variable (freqMult) that is equivalent to the amount of microseconds of a day times the amount of frequency between weeks / months / years. We then multiply this function based off of it's repeat type (weeks, months, years). For instance, if we have a task that repeats every two weeks, we first multiply it by the amount of microseconds in a day, then the frequency (2), then the repeat type (7).
2. After this we create another variable (numWeeks) that finds how many weeks the task should repeat.
3. We create a third variable (isValidFreqAway) we create is a bool variable that stores whether we are a valid frequency away from the start date
4.  We do a for loop that loops the amount of weeks the task is supposed to be active, and creates a new date that is equal to the next valid frequency. This is achieved by adding the (freqMult * (loop variable, in our case it's q) to our original date. We then mod this to the date we clicked on the calendar. If it's equal to 0, that means that the day is valid, and the isValidFreqAway turns true.
5. In an if statement that determines if a task should be added to the day pop up, if the date isn't equal to the date the original date was set, we check if it's week day name matches, the day we clicked on is not before the task was created, the day is not after the end date, and that isValidFreqAway == true.

Week frequencies work, but as of right now months and years have not been implemented, and I still haven't fixed it so when a user completes the task it doesn't complete all future ones.
